### PR TITLE
tests: move IT in `executor` to `tests/integrationtest` (PART 6)

### PR DESCRIPTION
**Pull request by Defined2014**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tidb/issues/47076

Problem Summary: Move 42 test cases.

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  >   

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by hawkingrei — APPROVED** [review]

File: tests/integrationtest/t/executor/index_lookup_merge_join.test [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+# TestIssue20549
+drop table if exists t1, t2;
+CREATE TABLE `t1` (`id` bigint(20) NOT NULL AUTO_INCREMENT, `t2id` bigint(20) DEFAULT NULL, PRIMARY KEY (`id`), KEY `t2id` (`t2id`));
+INSERT INTO `t1` VALUES (1,NULL);
+CREATE TABLE `t2` (`id` bigint(20) NOT NULL AUTO_INCREMENT, PRIMARY KEY (`id`));
+SELECT /*+ INL_MERGE_JOIN(t1,t2)  */ 1 from t1 left outer join t2 on t1.t2id=t2.id;
+SELECT /*+ HASH_JOIN(t1,t2)  */ 1 from t1 left outer join t2 on t1.t2id=t2.id;
+
+# TestIssue24473
+drop table if exists x, t2, t3;
```
Comment (SeaRise) [inline tests/integrationtest/t/executor/index_lookup_merge_join.test:92]:
```suggestion
drop table if exists x;
```

File: tests/integrationtest/t/executor/index_lookup_merge_join.test [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+CREATE TABLE `x` (  `a` enum('y','b','1','x','0','null') DEFAULT NULL,  KEY `a` (`a`));
+insert into x values("x"),("x"),("b"),("y");
+--sorted_result
+SELECT /*+ merge_join (t2,t3) */ t2.a,t3.a FROM x t2 inner join x t3 on t2.a = t3.a;
+--sorted_result
+SELECT /*+ inl_merge_join (t2,t3) */ t2.a,t3.a FROM x t2 inner join x t3 on t2.a = t3.a;
+
+# TestIssue25669
+drop table if exists x, t2, t3;
```
Comment (SeaRise) [inline tests/integrationtest/t/executor/index_lookup_merge_join.test:101]:
```suggestion
drop table if exists x;
```

File: tests/integrationtest/t/executor/infoschema_reader.test [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+select tidb_table_id > 0 from information_schema.tables where table_schema = 'executor__infoschema_reader' and table_name = 't';
+
+# TestSchemataCharacterSet
+drop database if exists `foo`;
+CREATE DATABASE `foo` DEFAULT CHARACTER SET = 'utf8mb4';
+select default_character_set_name, default_collation_name FROM information_schema.SCHEMATA  WHERE schema_name = 'foo';
+drop database `foo`;
+
+# TestViews
+CREATE DEFINER='root'@'localhost' VIEW executor__infoschema_reader.v1 AS SELECT 1;
```
Comment (SeaRise) [inline tests/integrationtest/t/executor/infoschema_reader.test:42]: drop view if exists first?

**Review by SeaRise — APPROVED** [review]

**Pull request merged**